src/async-wrap.cc

```diff
@@ -249,6 +259,16 @@ bool AsyncWrap::EmitBefore(Environment* env, double async_id) {
 
 
 static bool PostCallbackExecution(AsyncWrap* wrap, bool run_domain_cbs) {
+  switch (wrap->provider_type()) {
+#define V(PROVIDER)                                                           \
+    case AsyncWrap::PROVIDER_ ## PROVIDER:                                    \
+      TRACE_EVENT_NESTABLE_ASYNC_END0("node", #PROVIDER " cb",                \
+        static_cast<int64_t>(wrap->get_id()));                                \
+      break;
+    NODE_ASYNC_PROVIDER_TYPES(V)
+#undef V
+  }
```

Comment:
We could expose a check for whether trace events are enabled if necessary but each of these macros starts by running such a check and I would guess the overhead of the switch alone would be minimal.